packages/utils/src/carequality/cq-monthly-stats/athena-shared.ts

```diff
@@ -0,0 +1,225 @@
+    const output = await client.send(new GetQueryExecutionCommand(params));
+
+    if (
+      output &&
+      output.QueryExecution &&
+      output.QueryExecution.Status &&
+      output.QueryExecution.Status.State &&
+      output.QueryExecution.Status.State !== "SUCCEEDED"
+    ) {
+      console.log("Athena Query status is running");
+      return true;
+    }
+
+    console.log("Athena Query status is Succeeded");
+
+    return false;
+  } catch (error) {
+    return false;
+  }
+}
+
+async function getQueryResultByExecutionId(queryExecutionId: string): Promise<{
+  rows: Row[];
+  columns: ColumnInfo[];
+}> {
+  const params = {
+    QueryExecutionId: queryExecutionId,
+  };
+
+  let columns: ColumnInfo[] = [];
+  let resultsRows: Row[] = [];
+
+  const resp = await client.send(new GetQueryResultsCommand(params));
+
+  console.log("resp first:", resp.NextToken);
+  console.log("resp first length:", resp.ResultSet?.Rows?.length);
+
+  if (resp.NextToken) {
+    let nextToken: string | undefined = resp.NextToken;
+    resultsRows = resultsRows.concat(resp.ResultSet?.Rows ?? []);
+
+    while (nextToken) {
+      const nextParams: GetQueryResultsCommandInput = {
+        QueryExecutionId: queryExecutionId,
+        NextToken: nextToken,
+      };
+
+      const nextResp = await client.send(new GetQueryResultsCommand(nextParams));
+
+      console.log("resp second:", nextResp.NextToken);
+      console.log("resp second length:", nextResp.ResultSet?.Rows?.length);
+
+      resultsRows = resultsRows.concat(nextResp.ResultSet?.Rows ?? []);
+
+      nextToken = nextResp.NextToken;
+
+      sleep(5000);
+    }
+  } else {
+    resultsRows = resp.ResultSet?.Rows ?? [];
+  }
+
+  if (resp.ResultSet?.ResultSetMetadata?.ColumnInfo) {
+    columns = resp.ResultSet?.ResultSetMetadata?.ColumnInfo;
+  }
+
+  return {
+    rows: resultsRows,
+    columns: columns,
+  };
+}
+
+function buildTable(output: {
+  rows: Row[];
+  columns: ColumnInfo[];
+}): Array<{ [key: string]: string }> {
+  const columns = output.columns.map(column => column.Name) ?? [];
+
+  const rows = output.rows.map(row => {
+    return row.Data?.map(data => {
+      return data.VarCharValue;
+    });
+  });
+
+  rows?.shift();
+
+  const colToRow = rows?.map(row => {
+    return columns.reduce((acc, curr, index) => {
+      if (curr && row) {
+        return { ...acc, [curr]: row[index] };
+      }
+
+      return acc;
+    }, {});
+  });
+
+  if (!colToRow) {
+    return [];
+  }
+
+  return colToRow;
+}
+
+function addDurationsToTable(tableResults: Array<{ [key: string]: string }>): TableResults[] {
+  return tableResults.map(result => {
+    const requesttimestamp = dayjs(result.requesttimestamp);
+    const responsetimestamp = dayjs(result.responsetimestamp);
+
+    return {
+      ...result,
+      duration: responsetimestamp.diff(requesttimestamp, "milliseconds"),
+    };
+  }) as TableResults[];
+}
+
+export function getDurationsPerGW(results: TableResults[]): GWWithStats[] {
```

Comment:
This function is almost exactly the same as `getDurationsPerGW` from `./shared.ts`.

We could pass a function that returns the `gwId` given a single `result`.